From 5c5150a43a9a7ceff0187de1949af0ec9e597373 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 19:53:37 +0530
Subject: [PATCH 132/500] OpenWrt: 660-fq_codel_defaults.patch

hack: net: fq_codel: tune defaults for small devices

Assume that x86_64 devices always have a big memory and do not need this
optimization compared to devices with only 32 MB or 64 MB RAM.

Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: I7e7ea394bc53e439ce4d08997ff4a59e2300f429
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 51109b0f1f204441c4da95e52cd991e6ce352e7d)
---
 net/sched/sch_fq_codel.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/sched/sch_fq_codel.c b/net/sched/sch_fq_codel.c
index 8254495808b4..2c46208357da 100644
--- a/net/sched/sch_fq_codel.c
+++ b/net/sched/sch_fq_codel.c
@@ -471,7 +471,11 @@ static int fq_codel_init(struct Qdisc *sch, struct nlattr *opt,
 
 	sch->limit = 10*1024;
 	q->flows_cnt = 1024;
+#ifdef CONFIG_X86_64
 	q->memory_limit = 32 << 20; /* 32 MBytes */
+#else
+	q->memory_limit = 4 << 20; /* 4 MBytes */
+#endif
 	q->drop_batch_size = 64;
 	q->quantum = psched_mtu(qdisc_dev(sch));
 	INIT_LIST_HEAD(&q->new_flows);
-- 
2.34.1

